From 17cf9043b4700ed68d365d02093a223b6662e94d Mon Sep 17 00:00:00 2001
From: Zhong Jinghua <zhongjinghua@huawei.com>
Date: Mon, 21 Nov 2022 23:17:45 +0800
Subject: [PATCH 0083/1398] [Huawei] Revert "[Huawei] bitops: add a function:
 test_and_set_bit_relaxed()"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187991

----------------------------------------

This reverts commit 18698e7ee1e26a4b5180f84b4db2c36a174f99cc.

This patch doesn't actually fix hisi_sas_v3 performance issue.

Signed-off-by: Zhong Jinghua <zhongjinghua@huawei.com>
---
 include/asm-generic/bitops/atomic.h | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/include/asm-generic/bitops/atomic.h b/include/asm-generic/bitops/atomic.h
index 746dddde27aa..21aa26e7c988 100644
--- a/include/asm-generic/bitops/atomic.h
+++ b/include/asm-generic/bitops/atomic.h
@@ -29,21 +29,6 @@ static __always_inline void change_bit(unsigned int nr, volatile unsigned long *
 	atomic_long_xor(BIT_MASK(nr), (atomic_long_t *)p);
 }
 
-static inline int test_and_set_bit_relaxed(unsigned int nr, volatile unsigned long *p)
-{
-	long old;
-	unsigned long mask = BIT_MASK(nr);
-
-	p += BIT_WORD(nr);
-	if (READ_ONCE(*p) & mask)
-		return 1;
-
-	old = atomic_long_fetch_or(mask, (atomic_long_t *)p);
-	return !!(old & mask);
-}
-
-#define test_and_set_bit_relaxed test_and_set_bit_relaxed
-
 static inline int test_and_set_bit(unsigned int nr, volatile unsigned long *p)
 {
 	long old;
-- 
2.17.1

